_test/test_algorithms/test_cut.m

```diff
@@ -274,6 +277,88 @@ function test_error_raised_if_cut_called_with_multiple_files(obj)
         assertExceptionThrown(f, 'HORACE:cut');
     end
 
+    function test_you_can_take_an_out_of_memory_cut_with_tmp_files_with_mex(obj)
+        pix_pg_size = 5e5;  % this gives two pages of pixels over obj.sqw_file
+        use_mex = true;
+        outfile = fullfile(tmp_dir, 'tmp_outfile.sqw');
+        cleanup = obj.do_out_of_mem_cut(obj.sqw_file, outfile, pix_pg_size, use_mex);
+
+        ref_sqw = sqw(obj.ref_file);
+        output_sqw = sqw(outfile);
+
+        assertEqualToTol(output_sqw, ref_sqw, obj.FLOAT_TOL, 'ignore_str', true);
+    end
+
+    function test_you_can_take_an_out_of_memory_cut_with_tmp_files_no_mex(obj)
+        pix_pg_size = 5e5;  % this gives two pages of pixels over obj.sqw_file
+        use_mex = false;
+        outfile = fullfile(tmp_dir, 'tmp_outfile.sqw');
+        cleanup = obj.do_out_of_mem_cut(obj.sqw_file, outfile, pix_pg_size, use_mex);
```

Comment:
I'm not sure if the `do_out_of_mem_cut` helper is making the test clearer or more complex